fastly/resource_fastly_alert.go

```diff
@@ -102,9 +102,9 @@ func resourceFastlyAlert() *schema.Resource {
 
 			"service_id": {
 				Type:        schema.TypeString,
-				Required:    true,
+				Optional:    true,
 				ForceNew:    true,
-				Description: "The service which the alert monitors.",
+				Description: "The service which the alert monitors. Optional when using `stats` as the `source`.",
```

Comment:
We should have some code added to the provider to check if `source` is set to `domains` or `origins` and if so, and the `service_id` is not provided, then we error. Because my understanding is that `service_id` is _required_ if either of those other two values are set for `source`.

And have a test around that behaviour also.

Reply:
Sounds good

Reply:
And yes, you are correct.